### PR TITLE
Add Gemma2

### PR DESCRIPTION
**Pull request by Ying1123**
Logits aligned with Huggingface when using float16. Our default is bfloat16.

### CODE REVIEW

File: python/sglang/srt/layers/logits_processor.py [hunk 1]
```diff
@@ -108,6 +108,11 @@ def forward(self, input_ids, hidden_states, weight, input_metadata: InputMetadat
             last_logits = tensor_model_parallel_all_gather(last_logits)
         last_logits = last_logits[:, : self.config.vocab_size]
 
+        if hasattr(self.config, "final_logit_softcapping"):
+            last_logits = last_logits / self.config.final_logit_softcapping
```
Comment (merrymercy) [inline python/sglang/srt/layers/logits_processor.py]: use in-place operation

File: python/sglang/srt/layers/radix_attention.py [hunk 2]
```diff
@@ -21,40 +19,26 @@ def __init__(
...
-            o.view(-1, self.tp_q_head_num, self.head_dim),
-            input_metadata.start_loc,
-            input_metadata.seq_lens,
-            input_metadata.max_seq_len,
-            self.logit_cap,
-        )
-        self.store_kv_cache(k, v, input_metadata)
-
-        return o
+        # In SGLang, the typical prefill and prefill with cache are all merged into extend.
```
Comment (merrymercy) [inline python/sglang/srt/layers/radix_attention.py]:
```suggestion
        # In SGLang, we call both the typical "prefill" and "prefill with cache" as "extend".
        # See the extend_forward_xxx functions.
```

File: python/sglang/srt/models/gemma2.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+        use_sliding_window = (layer_idx % 2 == 1
+                              and config.sliding_window is not None)
+        del use_sliding_window  # Unused.
+        self.attn = RadixAttention(self.num_heads,
+                                   self.head_dim,
+                                   self.scaling,
+                                   num_kv_heads=self.num_kv_heads,
+                                   layer_id=layer_idx,
+                                   logit_cap=self.config.attn_logit_softcapping,
+                                  )
```
Comment (merrymercy) [inline python/sglang/srt/models/gemma2.py]:
```suggestion
                                   logit_cap=self.config.attn_logit_softcapping)
```

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/logits_processor.py [hunk 4]
```diff
@@ -109,9 +109,9 @@ def forward(self, input_ids, hidden_states, weight, input_metadata: InputMetadat
         last_logits = last_logits[:, : self.config.vocab_size]
 
         if hasattr(self.config, "final_logit_softcapping"):
-            last_logits = last_logits / self.config.final_logit_softcapping
+            last_logits /= self.config.final_logit_softcapping
             last_logits = torch.tanh(last_logits)
-            last_logits = last_logits * self.config.final_logit_softcapping
+            last_logits *= self.config.final_logit_softcapping
```
Comment (merrymercy) [inline python/sglang/srt/layers/logits_processor.py:114]:
```
            last_logits.mul_(self.config.final_logit_softcapping)
```

File: python/sglang/srt/layers/logits_processor.py [hunk 5]
```diff
@@ -109,9 +109,9 @@ def forward(self, input_ids, hidden_states, weight, input_metadata: InputMetadat
         last_logits = last_logits[:, : self.config.vocab_size]
 
         if hasattr(self.config, "final_logit_softcapping"):
-            last_logits = last_logits / self.config.final_logit_softcapping
+            last_logits /= self.config.final_logit_softcapping
```
Comment (merrymercy) [inline python/sglang/srt/layers/logits_processor.py:112]:
```
            last_logits.div_(self.config.final_logit_softcapping)
```
